File: backend/src/main/java/com/x/pojo/entity/Tool.java
package com.x.pojo.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Tool implements Serializable {
    private Long id;
    private String name;
    private String url;
    private String description;
    private String avatar;
    private String cardColor;
}

File: frontend/src/api/tool.ts
import request from "@/utils/request";

export interface Tool{
    id: number
  name: string
  url: string
  description: string
  avatar: string
  cardColor: string
}

export const getTools= async()=>{
    try {
        const res=await request.get('/user/tool/getAll')
        if(res.data.code === 1){
            return res.data.data as Tool[]
        }else{
            throw new Error(res.data.msg || '获取工具列表失败')
        }
    }
    catch (error) {
        throw error
    }
}

export const add=async (tool:Tool)=>{
    try{
        const res=await request.post('/user/tool/add',tool)
        if(res.data.code!= 1)
        {
            throw new Error(res.data.msg || '添加工具失败')
        }
        return res.data.data
    }
    catch(error){
        console.log(error)
        throw error
    }
}